src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs

```diff
@@ -202,25 +197,79 @@ private static OtlpTrace.Span.Types.Event ToOtlpEvent(ActivityEvent activityEven
             return otlpEvent;
         }
 
+        private static OtlpCommon.AttributeKeyValue ToOtlpAttribute(KeyValuePair<string, string> kvp)
+        {
+            // TODO: enforce no duplicate keys?

```

Comment:
.NET team if making Activity.Attributes to match OT spec - so the order will be maintained, duplicated will be prevented etc. 

https://github.com/dotnet/designs/pull/134/files#diff-1f85dcaf1be8ff277864f69c473c6b33R34

Reply:
Great. This logic will need to be touched again once those activity changes are made by them. Looks like we will be able to utilize the Activity.Attributes when it becomes available

Reply:
Yes.